Kernel/Process.cpp

```diff
@@ -4048,28 +4052,24 @@ int Process::sys$mount(const Syscall::SC_mount_params* user_params)
 
     if (params.flags & MS_BIND) {
         // We're doing a bind mount.
-        auto source_or_error = VFS::the().resolve_path(source, current_directory());
-        if (source_or_error.is_error())
-            return source_or_error.error();
-        auto& source_custody = source_or_error.value();
-        return VFS::the().bind_mount(source_custody, target_custody, params.flags);
+        if (fd.is_null())
+            return -EBADF;
+        ASSERT(fd->custody());
+        return VFS::the().bind_mount(*fd->custody(), target_custody, params.flags);
     }
 
     if (fs_type == "ext2" || fs_type == "Ext2FS") {
-        auto source_or_error = VFS::the().open(source, O_RDWR, 0, current_directory());
-        if (source_or_error.is_error())
-            return source_or_error.error();
-
-        auto* device = source_or_error.value()->device();
-        if (!device || !device->is_block_device()) {
-            dbg() << "mount: this is not a BlockDevice";
+        if (fd.is_null())
+            return -EBADF;
+        ASSERT(fd->custody());
+        if (!fd->file().is_seekable()) {
+            dbg() << "mount: this is not a seekable file";
```

Comment:
Great; that's what i meant by checking upfront. But let's also add an assertion in the `FBFS` constructor.